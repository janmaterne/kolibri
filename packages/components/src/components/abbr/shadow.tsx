import type { JSX } from '@stencil/core';
import { Component, h, Host, Prop, State, Watch } from '@stencil/core';
import type { AbbrAPI, AbbrStates, LabelPropType } from '../../schema';
import { validateLabel } from '../../schema';

/**
 * @slot - The abbreviation (short form).
 */
@Component({
	tag: 'kol-abbr',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolAbbr implements AbbrAPI {
	public render(): JSX.Element {
		return (
			<Host class="kol-abbr">
				<abbr>
					<slot />
				</abbr>
				{this.state._label ? ` (${this.state._label})` : ''}
			</Host>
		);
	}

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label?: LabelPropType;

	@State() public state: AbbrStates = {
		_label: '', // âš  required
	};

	@Watch('_label')
	public validateLabel(value?: LabelPropType): void {
		validateLabel(this, value, {
			required: true,
		});
	}

	public componentWillLoad(): void {
		this.validateLabel(this._label);
	}
}
