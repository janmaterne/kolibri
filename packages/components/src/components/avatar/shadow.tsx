import type { JSX } from '@stencil/core';
import { Component, h, Prop } from '@stencil/core';

import type { AvatarProps } from '../../schema';
import { KolAvatarWcTag } from '../../core/component-names';

@Component({
	tag: 'kol-avatar',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolAvatar implements AvatarProps {
	public render(): JSX.Element {
		return <KolAvatarWcTag _src={this._src} _label={this._label}></KolAvatarWcTag>;
	}

	/**
	 * Sets the image `src` attribute to the given string.
	 */
	@Prop() public _src?: string;

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label!: string;
}
