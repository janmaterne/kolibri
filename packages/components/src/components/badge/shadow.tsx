import type { BadgeAPI, BadgeStates, ButtonProps, KoliBriIconsProp, LabelPropType, PropColor, Stringified } from '../../schema';
import { featureHint, handleColorChange, objectObjectHandler, parseJson, setState, validateColor, validateIcons } from '../../schema';
import { Component, h, Host, Prop, State, Watch } from '@stencil/core';
import { KolSpanFc } from '@fc';

import { nonce } from '../../utils/dev.utils';

import type { JSX } from '@stencil/core';
import { KolButtonWcTag } from '../../core/component-names';
featureHint(`[KolBadge] Optimierung des _color-Properties (rgba, rgb, hex usw.).`);

@Component({
	tag: 'kol-badge',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolBadge implements BadgeAPI {
	private bgColorStr = '#000';
	private colorStr = '#fff';
	private readonly id = nonce();

	private renderSmartButton(props: ButtonProps): JSX.Element {
		return (
			<KolButtonWcTag
				_ariaControls={this.id}
				_customClass={props._customClass}
				_disabled={props._disabled}
				_hideLabel={true}
				_icons={props._icons}
				_id={props._id}
				_label={props._label}
				_on={props._on}
				_tooltipAlign={props._tooltipAlign}
				_variant={props._variant}
			></KolButtonWcTag>
		);
	}

	public render(): JSX.Element {
		const hasSmartButton = typeof this.state._smartButton === 'object' && this.state._smartButton !== null;

		return (
			<Host class="kol-badge">
				<span
					class={{
						'smart-button': typeof this.state._smartButton === 'object' && this.state._smartButton !== null,
					}}
					style={{
						backgroundColor: this.bgColorStr,
						color: this.colorStr,
					}}
				>
					<KolSpanFc id={hasSmartButton ? this.id : undefined} allowMarkdown icons={this.state._icons} label={this._label} />
					{hasSmartButton && this.renderSmartButton(this.state._smartButton as ButtonProps)}
				</span>
			</Host>
		);
	}

	/**
	 * Defines the backgroundColor and foregroundColor.
	 */
	@Prop() public _color?: Stringified<PropColor> = '#000';

	/**
	 * Defines the icon classnames (e.g. `_icons="fa-solid fa-user"`).
	 */
	@Prop() public _icons?: Stringified<KoliBriIconsProp>;

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label!: LabelPropType;

	/**
	 * Allows to add a button with an arbitrary action within the element (_hide-label only).
	 */
	@Prop() public _smartButton?: Stringified<ButtonProps>;

	@State() public state: BadgeStates = {
		_color: {
			backgroundColor: '#000',
			foregroundColor: '#fff',
		},
		_icons: {},
	};

	private handleColorChange = (value: unknown) => {
		const colorPair = handleColorChange(value);
		this.bgColorStr = colorPair.backgroundColor;
		this.colorStr = colorPair.foregroundColor as string;
	};

	@Watch('_icons')
	public validateIcons(value?: KoliBriIconsProp): void {
		validateIcons(this, value);
	}

	@Watch('_color')
	public validateColor(value?: Stringified<PropColor>): void {
		validateColor(this, value, {
			defaultValue: '#000',
			hooks: {
				beforePatch: this.handleColorChange,
			},
		});
	}

	@Watch('_smartButton')
	public validateSmartButton(value?: ButtonProps | string): void {
		objectObjectHandler(value, () => {
			try {
				value = parseJson<ButtonProps>(value as string);
				// eslint-disable-next-line no-empty
			} catch (e) {
				// value behält den ursprünglichen Wert
			}
			setState(this, '_smartButton', value);
		});
	}

	public componentWillLoad(): void {
		this.validateIcons(this._icons);
		this.validateColor(this._color);
		this.validateSmartButton(this._smartButton);
	}
}
