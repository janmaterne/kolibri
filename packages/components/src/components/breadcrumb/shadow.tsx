import type { BreadcrumbAPI, BreadcrumbLinkProps, BreadcrumbStates, LabelPropType, LinkProps, Stringified } from '../../schema';
import { a11yHintLabelingLandmarks, validateLabel } from '../../schema';
import { Component, Fragment, h, Prop, State, Watch } from '@stencil/core';

import { addNavLabel, removeNavLabel } from '../../utils/unique-nav-labels';
import { watchNavLinks } from '../nav/validation';

import type { JSX } from '@stencil/core';
import { KolIconTag, KolLinkTag } from '../../core/component-names';

@Component({
	tag: 'kol-breadcrumb',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolBreadcrumb implements BreadcrumbAPI {
	private readonly renderLink = (link: BreadcrumbLinkProps, index: number): JSX.Element => {
		const lastIndex = this.state._links.length - 1;
		return (
			<li class="kol-breadcrumb__list-element" key={index}>
				{index !== 0 && <KolIconTag class="kol-breadcrumb__icon" _label="" _icons="codicon codicon-chevron-right" />}
				{index === lastIndex ? (
					<span class="kol-breadcrumb__list-element-span">
						{link._hideLabel ? (
							<KolIconTag
								class="kol-breadcrumb__icon"
								_label={link._label}
								_icons={typeof link._icons === 'string' ? link._icons : 'codicon codicon-symbol-event'}
							/>
						) : (
							<>{link._label}</>
						)}
					</span>
				) : (
					<KolLinkTag class="kol-breadcrumb__link" {...link}></KolLinkTag>
				)}
			</li>
		);
	};

	public render(): JSX.Element {
		return (
			<nav class="kol-breadcrumb" aria-label={this.state._label}>
				<ul class="kol-breadcrumb__list">
					{this.state._links.length === 0 && (
						<li>
							<KolIconTag class="kol-breadcrumb_icon" _label="" _icons="codicon codicon-home" />…
						</li>
					)}
					{this.state._links.map(this.renderLink)}
				</ul>
			</nav>
		);
	}

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label!: LabelPropType;

	/**
	 * Defines the list of links combined with their labels to render.
	 */
	@Prop() public _links!: Stringified<BreadcrumbLinkProps[]>;

	@State() public state: BreadcrumbStates = {
		_label: '', // ⚠ required
		_links: [],
	};

	@Watch('_label')
	public validateLabel(value?: LabelPropType, _oldValue?: LabelPropType, initial = false): void {
		if (!initial) {
			removeNavLabel(this.state._label); // remove the current
		}
		validateLabel(this, value, {
			required: true,
		});
		a11yHintLabelingLandmarks(value);
		addNavLabel(this.state._label); // add the state instead of prop, because the prop could be invalid and not set as new label
	}

	@Watch('_links')
	public validateLinks(value?: Stringified<LinkProps[]>): void {
		watchNavLinks('KolBreadcrumb', this, value);
	}

	public componentWillLoad(): void {
		this.validateLabel(this._label, undefined, true);
		this.validateLinks(this._links);
	}

	public disconnectedCallback(): void {
		removeNavLabel(this.state._label);
	}
}
