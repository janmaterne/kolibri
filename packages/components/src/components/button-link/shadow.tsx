import { Component, Element, h, Host, JSX, Prop } from '@stencil/core';

import { Stringified } from '../../types/common';
import { AlternativeButtonLinkRolePropType } from '../../types/props/alternative-button-link-role';
import { AriaCurrentPropType } from '../../types/props/aria-current';
import { ButtonCallbacksPropType } from '../../types/props/button-callbacks';
import { ButtonTypePropType } from '../../types/props/button-type';
import { IconsPropType } from '../../types/props/icons';
import { IdPropType } from '../../types/props/id';
import { LabelWithExpertSlotPropType } from '../../types/props/label';
import { NamePropType } from '../../types/props/name';
import { SyncValueBySelectorPropType } from '../../types/props/sync-value-by-selector';
import { TooltipAlignPropType } from '../../types/props/tooltip-align';
import { StencilUnknown } from '../../types/unknown';
import { propagateFocus } from '../../utils/reuse';
import { Props } from './types';
import { KolButtonWcTag } from '../../core/component-names';

@Component({
	tag: 'kol-button-link',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolButtonLink implements Props {
	@Element() private readonly host?: HTMLKolButtonLinkElement;

	private readonly catchRef = (ref?: HTMLKolButtonWcElement) => {
		propagateFocus(this.host, ref);
	};

	public render(): JSX.Element {
		return (
			<Host class="kol-button-link">
				<KolButtonWcTag
					ref={this.catchRef}
					_accessKey={this._accessKey}
					_ariaControls={this._ariaControls}
					_ariaCurrent={this._ariaCurrent}
					_ariaExpanded={this._ariaExpanded}
					_ariaLabel={this._ariaLabel}
					_ariaSelected={this._ariaSelected}
					_disabled={this._disabled}
					_icons={this._icons || this._icon}
					_hideLabel={this._hideLabel || this._iconOnly}
					_id={this._id}
					_label={this._label}
					_name={this._name}
					_on={this._on}
					_role="link"
					_syncValueBySelector={this._syncValueBySelector}
					_tabIndex={this._tabIndex}
					_tooltipAlign={this._tooltipAlign}
					_type={this._type}
					_value={this._value}
				>
					<slot name="expert" slot="expert"></slot>
				</KolButtonWcTag>
			</Host>
		);
	}

	/**
	 * Defines which key combination can be used to trigger or focus the interactive element of the component.
	 */
	@Prop() public _accessKey?: string;

	/**
	 * Defines which elements are controlled by this component. (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls)
	 */
	@Prop() public _ariaControls?: string;

	/**
	 * Deprecated: Marks the element as the selected in a group of related elements. Can be one of the following: `date` | `location` | `page` | `step` | `time` | `true`. (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current)
	 *
	 * @deprecated aria-current is not necessary for buttons. will be removed in version 2.
	 */
	@Prop() public _ariaCurrent?: AriaCurrentPropType;

	/**
	 * Defines whether the interactive element of the component expanded something. (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded)
	 * @TODO: Change type back to `AriaExpandedPropType` after Stencil#4663 has been resolved.
	 */
	@Prop() public _ariaExpanded?: boolean;

	/**
	 * Deprecated: Setzt die semantische Beschriftung der Komponente.
	 *
	 * @deprecated use _label instead
	 */
	@Prop({ mutable: true, reflect: false }) public _ariaLabel?: string;

	/**
	 * Defines whether the interactive element of the component is selected (e.g. role=tab). (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-selected)
	 * @TODO: Change type back to `AriaSelectedPropType` after Stencil#4663 has been resolved.
	 */
	@Prop() public _ariaSelected?: boolean;

	/**
	 * Makes the element not focusable and ignore all events.
	 * @TODO: Change type back to `DisabledPropType` after Stencil#4663 has been resolved.
	 */
	@Prop() public _disabled?: boolean = false;

	/**
	 * Hides the caption by default and displays the caption text with a tooltip when the
	 * interactive element is focused or the mouse is over it.
	 * @TODO: Change type back to `HideLabelPropType` after Stencil#4663 has been resolved.
	 */
	@Prop() public _hideLabel?: boolean = false;

	/**
	 * @deprecated Use _icons.
	 */
	@Prop() public _icon?: IconsPropType;

	/**
	 * Defines the icon classnames (e.g. `_icons="fa-solid fa-user"`).
	 */
	@Prop() public _icons?: IconsPropType;

	/**
	 * Deprecated: Hides the label and shows the description in a Tooltip instead.
	 * @deprecated use _hide-label
	 */
	@Prop() public _iconOnly?: boolean;

	/**
	 * Defines the internal ID of the primary component element.
	 */
	@Prop() public _id?: IdPropType;

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.). Set to `false` to enable the expert slot.
	 */
	@Prop() public _label!: LabelWithExpertSlotPropType;

	/**
	 * Defines the technical name of an input field.
	 */
	@Prop() public _name?: NamePropType;

	/**
	 * Gibt die EventCallback-Funktionen f√ºr die Button-Events an.
	 */
	@Prop() public _on?: ButtonCallbacksPropType<StencilUnknown>;

	/**
	 * Defines the role of the components primary element.
	 */
	@Prop() public _role?: AlternativeButtonLinkRolePropType;

	/**
	 * Selector for synchronizing the value with another input element.
	 * @internal
	 */
	@Prop() public _syncValueBySelector?: SyncValueBySelectorPropType;

	/**
	 * Defines which tab-index the primary element of the component has. (https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex)
	 */
	@Prop() public _tabIndex?: number;

	/**
	 * Defines where to show the Tooltip preferably: top, right, bottom or left.
	 */
	@Prop() public _tooltipAlign?: TooltipAlignPropType = 'top';

	/**
	 * Defines either the type of the component or of the components interactive element.
	 */
	@Prop() public _type?: ButtonTypePropType = 'button';

	/**
	 * Defines the value that the button emits on click.
	 */
	@Prop() public _value?: Stringified<StencilUnknown>;
}
