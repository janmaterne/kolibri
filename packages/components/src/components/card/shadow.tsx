import { Component, h, Host, Prop, State, Watch } from '@stencil/core';
import type { JSX } from '@stencil/core';
import type { CardAPI, CardStates, HasCloserPropType, HeadingLevel, KoliBriAlertEventCallbacks, KoliBriCardEventCallbacks, LabelPropType } from '../../schema';
import { setState, validateHasCloser, validateLabel } from '../../schema';
import { watchHeadingLevel } from '../heading/validation';
import { KolCardFc } from '../../functional-components';

/**
 * @slot - Ermöglicht das Einfügen beliebigen HTML's in den Inhaltsbereich der Card.
 */
@Component({
	tag: 'kol-card',
	styleUrls: {
		default: './style.scss',
	},
	shadow: true,
})
export class KolCard implements CardAPI {
	private readonly handleCloseClick = () => {
		this._on?.onClose?.(new Event('Close'));
	};

	public render(): JSX.Element {
		const { _label: label, _level: level, _hasCloser: hasCloser } = this.state;

		return (
			<Host class="kol-card">
				<KolCardFc hasCloser={hasCloser} HeadingProps={{ level, label, onClick: this.handleCloseClick }}>
					<slot />
				</KolCardFc>
			</Host>
		);
	}

	/**
	 * Defines the event callback functions for the component.
	 */
	@Prop() public _on?: KoliBriCardEventCallbacks;

	/**
	 * Defines whether the element can be closed.
	 * @TODO: Change type back to `HasCloserPropType` after Stencil#4663 has been resolved.
	 */
	@Prop() public _hasCloser?: boolean = false;

	/**
	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
	 */
	@Prop() public _label!: LabelPropType;

	/**
	 * Defines which H-level from 1-6 the heading has. 0 specifies no heading and is shown as bold text.
	 */
	@Prop() public _level?: HeadingLevel = 1;

	@State() public state: CardStates = {
		_label: '', // ⚠ required
	};

	private validateOnValue = (value: unknown): boolean =>
		typeof value === 'object' && value !== null && typeof (value as KoliBriCardEventCallbacks).onClose === 'function';

	@Watch('_on')
	public validateOn(value?: KoliBriCardEventCallbacks): void {
		if (this.validateOnValue(value)) {
			setState<KoliBriCardEventCallbacks>(this, '_on', {
				onClose: (value as KoliBriAlertEventCallbacks).onClose,
			});
		}
	}

	@Watch('_hasCloser')
	public validateHasCloser(value?: HasCloserPropType): void {
		validateHasCloser(this, value);
	}

	@Watch('_label')
	public validateLabel(value?: LabelPropType): void {
		validateLabel(this, value, {
			required: true,
		});
	}

	@Watch('_level')
	public validateLevel(value?: HeadingLevel): void {
		watchHeadingLevel(this, value);
	}

	public componentWillLoad(): void {
		this.validateHasCloser(this._hasCloser);
		this.validateLabel(this._label);
		this.validateLevel(this._level);
		this.validateOn(this._on);
	}
}
