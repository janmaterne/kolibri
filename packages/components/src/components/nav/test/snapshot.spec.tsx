import { KolNavTag } from '@component-names';
import type { NavProps } from '@schema';
import { executeSnapshotTests } from '@testing';

import { KolNav } from '../shadow';

const baseObj: NavProps = {
	_label: 'Nav Label',
	_links: [
		{
			_label: 'Homepage',
			_href: '#',
		},
		{
			_label: 'Nav - aria-current',
			_href: '#',
			_active: true,
		},
		{
			_label: '3 Navigation point',
			_href: '#',
			_icons: 'codicon codicon-home',

			_children: [
				{
					_label: '3.1 Navigation point',
					_icons: 'codicon codicon-home',
					_href: '#',
				},
			],
		},

		{
			_label: '6 Keine eigene Seite, mit Icon',
			_icons: 'codicon codicon-squirrel',
		},
	],
};

executeSnapshotTests<NavProps>(
	KolNavTag,
	[KolNav],
	[
		{ ...baseObj },
		{ ...baseObj, _hasIconsWhenExpanded: false },
		{ ...baseObj, _hasIconsWhenExpanded: true },
		{ ...baseObj, _hideLabel: false },
		{ ...baseObj, _hideLabel: true },
		{ ...baseObj, _hasCompactButton: false },
		{ ...baseObj, _hasCompactButton: true },
	],
);
