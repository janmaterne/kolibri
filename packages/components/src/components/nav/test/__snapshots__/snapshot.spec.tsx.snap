// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kol-nav should render with _label="Nav Label" _links=[{"_label":"Homepage","_href":"#"},{"_label":"Nav - aria-current","_href":"#","_active":true},{"_label":"3 Navigation point","_href":"#","_icons":"codicon codicon-home","_children":[{"_label":"3.1 Navigation point","_icons":"codicon codicon-home","_href":"#"}]},{"_label":"6 Keine eigene Seite, mit Icon","_icons":"codicon codicon-squirrel"}] _hasCompactButton=false 1`] = `
<kol-nav class="kol-nav">
  <template shadowrootmode="open">
    <div class="nav vertical">
      <nav aria-label="Nav Label" id="nav">
        <ul class="list vertical" data-deep="0">
          <li>
            <div class="entry">
              <kol-link-wc _href="#" _label="Homepage" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="active">
            <div class="entry">
              <kol-link-wc _active="" _href="#" _label="Nav - aria-current" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="has-children">
            <div class="entry">
              <kol-link-wc _href="#" _label="3 Navigation point" class="entry-item"></kol-link-wc>
              <kol-button-wc _hidelabel="" _icons="codicon codicon-add" _label="kol-nav-label-open" class="expand-button"></kol-button-wc>
            </div>
          </li>
          <li>
            <div class="entry">
              <kol-button-wc _label="6 Keine eigene Seite, mit Icon" class="entry-item"></kol-button-wc>
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </template>
</kol-nav>
`;

exports[`kol-nav should render with _label="Nav Label" _links=[{"_label":"Homepage","_href":"#"},{"_label":"Nav - aria-current","_href":"#","_active":true},{"_label":"3 Navigation point","_href":"#","_icons":"codicon codicon-home","_children":[{"_label":"3.1 Navigation point","_icons":"codicon codicon-home","_href":"#"}]},{"_label":"6 Keine eigene Seite, mit Icon","_icons":"codicon codicon-squirrel"}] _hasCompactButton=true 1`] = `
<kol-nav class="kol-nav">
  <template shadowrootmode="open">
    <div class="nav vertical">
      <nav aria-label="Nav Label" id="nav">
        <ul class="list vertical" data-deep="0">
          <li>
            <div class="entry">
              <kol-link-wc _href="#" _label="Homepage" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="active">
            <div class="entry">
              <kol-link-wc _active="" _href="#" _label="Nav - aria-current" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="has-children">
            <div class="entry">
              <kol-link-wc _href="#" _label="3 Navigation point" class="entry-item"></kol-link-wc>
              <kol-button-wc _hidelabel="" _icons="codicon codicon-add" _label="kol-nav-label-open" class="expand-button"></kol-button-wc>
            </div>
          </li>
          <li>
            <div class="entry">
              <kol-button-wc _label="6 Keine eigene Seite, mit Icon" class="entry-item"></kol-button-wc>
            </div>
          </li>
        </ul>
      </nav>
      <div class="compact">
        <kol-button _ariacontrols="nav" _ariaexpanded="" _hidelabel="" _icons="codicon codicon-chevron-left" _label="kol-nav-minimize" _tooltipalign="right" _variant="ghost"></kol-button>
      </div>
    </div>
  </template>
</kol-nav>
`;

exports[`kol-nav should render with _label="Nav Label" _links=[{"_label":"Homepage","_href":"#"},{"_label":"Nav - aria-current","_href":"#","_active":true},{"_label":"3 Navigation point","_href":"#","_icons":"codicon codicon-home","_children":[{"_label":"3.1 Navigation point","_icons":"codicon codicon-home","_href":"#"}]},{"_label":"6 Keine eigene Seite, mit Icon","_icons":"codicon codicon-squirrel"}] _hasIconsWhenExpanded=false 1`] = `
<kol-nav class="kol-nav">
  <template shadowrootmode="open">
    <div class="nav vertical">
      <nav aria-label="Nav Label" id="nav">
        <ul class="list vertical" data-deep="0">
          <li>
            <div class="entry">
              <kol-link-wc _href="#" _label="Homepage" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="active">
            <div class="entry">
              <kol-link-wc _active="" _href="#" _label="Nav - aria-current" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="has-children">
            <div class="entry">
              <kol-link-wc _href="#" _label="3 Navigation point" class="entry-item"></kol-link-wc>
              <kol-button-wc _hidelabel="" _icons="codicon codicon-add" _label="kol-nav-label-open" class="expand-button"></kol-button-wc>
            </div>
          </li>
          <li>
            <div class="entry">
              <kol-button-wc _label="6 Keine eigene Seite, mit Icon" class="entry-item"></kol-button-wc>
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </template>
</kol-nav>
`;

exports[`kol-nav should render with _label="Nav Label" _links=[{"_label":"Homepage","_href":"#"},{"_label":"Nav - aria-current","_href":"#","_active":true},{"_label":"3 Navigation point","_href":"#","_icons":"codicon codicon-home","_children":[{"_label":"3.1 Navigation point","_icons":"codicon codicon-home","_href":"#"}]},{"_label":"6 Keine eigene Seite, mit Icon","_icons":"codicon codicon-squirrel"}] _hasIconsWhenExpanded=true 1`] = `
<kol-nav class="kol-nav">
  <template shadowrootmode="open">
    <div class="nav vertical">
      <nav aria-label="Nav Label" id="nav">
        <ul class="list vertical" data-deep="0">
          <li>
            <div class="entry">
              <kol-link-wc _href="#" _label="Homepage" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="active">
            <div class="entry">
              <kol-link-wc _active="" _href="#" _label="Nav - aria-current" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="has-children">
            <div class="entry">
              <kol-link-wc _href="#" _icons="codicon codicon-home" _label="3 Navigation point" class="entry-item"></kol-link-wc>
              <kol-button-wc _hidelabel="" _icons="codicon codicon-add" _label="kol-nav-label-open" class="expand-button"></kol-button-wc>
            </div>
          </li>
          <li>
            <div class="entry">
              <kol-button-wc _icons="codicon codicon-squirrel" _label="6 Keine eigene Seite, mit Icon" class="entry-item"></kol-button-wc>
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </template>
</kol-nav>
`;

exports[`kol-nav should render with _label="Nav Label" _links=[{"_label":"Homepage","_href":"#"},{"_label":"Nav - aria-current","_href":"#","_active":true},{"_label":"3 Navigation point","_href":"#","_icons":"codicon codicon-home","_children":[{"_label":"3.1 Navigation point","_icons":"codicon codicon-home","_href":"#"}]},{"_label":"6 Keine eigene Seite, mit Icon","_icons":"codicon codicon-squirrel"}] _hideLabel=false 1`] = `
<kol-nav class="kol-nav">
  <template shadowrootmode="open">
    <div class="nav vertical">
      <nav aria-label="Nav Label" id="nav">
        <ul class="list vertical" data-deep="0">
          <li>
            <div class="entry">
              <kol-link-wc _href="#" _label="Homepage" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="active">
            <div class="entry">
              <kol-link-wc _active="" _href="#" _label="Nav - aria-current" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="has-children">
            <div class="entry">
              <kol-link-wc _href="#" _label="3 Navigation point" class="entry-item"></kol-link-wc>
              <kol-button-wc _hidelabel="" _icons="codicon codicon-add" _label="kol-nav-label-open" class="expand-button"></kol-button-wc>
            </div>
          </li>
          <li>
            <div class="entry">
              <kol-button-wc _label="6 Keine eigene Seite, mit Icon" class="entry-item"></kol-button-wc>
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </template>
</kol-nav>
`;

exports[`kol-nav should render with _label="Nav Label" _links=[{"_label":"Homepage","_href":"#"},{"_label":"Nav - aria-current","_href":"#","_active":true},{"_label":"3 Navigation point","_href":"#","_icons":"codicon codicon-home","_children":[{"_label":"3.1 Navigation point","_icons":"codicon codicon-home","_href":"#"}]},{"_label":"6 Keine eigene Seite, mit Icon","_icons":"codicon codicon-squirrel"}] _hideLabel=true 1`] = `
<kol-nav class="kol-nav">
  <template shadowrootmode="open">
    <div class="is-compact nav vertical">
      <nav aria-label="Nav Label" id="nav">
        <ul class="list vertical" data-deep="0">
          <li>
            <div class="entry hide-label">
              <kol-link-wc _hidelabel="" _href="#" _icons="codicon codicon-symbol-method" _label="Homepage" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="active">
            <div class="entry hide-label">
              <kol-link-wc _active="" _hidelabel="" _href="#" _icons="codicon codicon-symbol-method" _label="Nav - aria-current" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="has-children">
            <div class="entry hide-label">
              <kol-link-wc _hidelabel="" _href="#" _icons="codicon codicon-home" _label="3 Navigation point" class="entry-item"></kol-link-wc>
              <kol-button-wc _hidelabel="" _icons="codicon codicon-add" _label="kol-nav-label-open" class="expand-button"></kol-button-wc>
            </div>
          </li>
          <li>
            <div class="entry hide-label">
              <kol-button-wc _hidelabel="" _icons="codicon codicon-squirrel" _label="6 Keine eigene Seite, mit Icon" class="entry-item"></kol-button-wc>
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </template>
</kol-nav>
`;

exports[`kol-nav should render with _label="Nav Label" _links=[{"_label":"Homepage","_href":"#"},{"_label":"Nav - aria-current","_href":"#","_active":true},{"_label":"3 Navigation point","_href":"#","_icons":"codicon codicon-home","_children":[{"_label":"3.1 Navigation point","_icons":"codicon codicon-home","_href":"#"}]},{"_label":"6 Keine eigene Seite, mit Icon","_icons":"codicon codicon-squirrel"}] 1`] = `
<kol-nav class="kol-nav">
  <template shadowrootmode="open">
    <div class="nav vertical">
      <nav aria-label="Nav Label" id="nav">
        <ul class="list vertical" data-deep="0">
          <li>
            <div class="entry">
              <kol-link-wc _href="#" _label="Homepage" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="active">
            <div class="entry">
              <kol-link-wc _active="" _href="#" _label="Nav - aria-current" class="entry-item"></kol-link-wc>
            </div>
          </li>
          <li class="has-children">
            <div class="entry">
              <kol-link-wc _href="#" _label="3 Navigation point" class="entry-item"></kol-link-wc>
              <kol-button-wc _hidelabel="" _icons="codicon codicon-add" _label="kol-nav-label-open" class="expand-button"></kol-button-wc>
            </div>
          </li>
          <li>
            <div class="entry">
              <kol-button-wc _label="6 Keine eigene Seite, mit Icon" class="entry-item"></kol-button-wc>
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </template>
</kol-nav>
`;
